src/NuGet.Clients/PackageManagement.UI/Feeds/MultiSourcePackageMetadataProvider.cs

```
@@ -204,8 +204,21 @@ internal class MultiSourcePackageMetadataProvider : IPackageMetadataProvider
                 .Select(task => task.ContinueWith(LogError, TaskContinuationOptions.OnlyOnFaulted))
                 .ToArray();
 
-            var completed = (await Task.WhenAll(tasks))
-                .Where(m => m != null);
+            IEnumerable<IPackageSearchMetadata> completed = null;
+            try
+            {
+                completed = (await Task.WhenAll(tasks))
```

if this throws does it still wait for all tasks to complete?

---

it might be easier to just do a foreach on the tasks, await each yourself in a try/catch and append the result to a list

---

If the idea is to ignore errors here then I'd suggest creating "safe" individual tasks
```cs
try
{
  return r.GetPackageMetadataAsync(identity, includePrerelease, cancellationToken);
}
catch
{
  return null;
}```